Fixes from a1 to a10

angelraver opened the pull request.

angelraver committed:
A1 - 1 Server Side JS Injection

angelraver committed:
A1 - 2 NoSQL Injection

angelraver committed:
A1 - 3 Log Injection

angelraver committed:
A2 - Broken Auth

angelraver committed:
A2 - Broken Auth and Session Management

angelraver committed:
A2 - Broken Auth requires stronger password

angelraver committed:
A3 - XSS

angelraver committed:
A4 - Insecure DOR

angelraver committed:
A5 - Security MisConfig

angelraver committed:
A6 - Sensitive Data Exposure

angelraver committed:
A7-Missing Function Level Access Control

angelraver committed:
A8 - Cross-Site Request Forgery (CSRF)

angelraver committed:
Update README.md

angelraver committed:
Fix for A9 - Insecure Dependencies

nbarrera commented on server.js:
```diff
@@ -82,26 +79,20 @@ MongoClient.connect(db, (err, db) => {
         cookie: {
             httpOnly: true
             // Remember to start an HTTPS server to get this working
             // secure: true
```
che aca yo le habilite

```
            secure: true,
            maxAge: 86400000 // 1 day in ms
```

nbarrera commented on app/routes/session.js:
```diff
@@ -55,39 +55,35 @@ function SessionHandler(db) {
-            const invalidPasswordErrorMessage = "Invalid password";
             if (err) {
                 if (err.noSuchUser) {
                     console.log('Error: attempt to login with invalid user: ', userName);
```
este es el codigo vulnerable al log injection creo, habria que borrarlo
```suggestion
```

nbarrera committed:
Merge branch 'fixes-from-A1-to-A10' of https://github.com/angelraver/desarrollo-seguro-grupo-2 into fixes-from-A1-to-A10

nbarrera reviewed (approved):
esta joya..
puse algunos comments / suggestions
y tengo un par de cambios mas para meter que tienen que ver con que funque cuando lo levantas

nbarrera commented:
En el archivo `app/views/login.html` meto este cambio para que el script ese no aparezca escapeado y se vea en el html en la pantalla de login

```diff
--- a/app/views/login.html
+++ b/app/views/login.html
@@ -143,7 +143,7 @@
     <script src="/vendor/bootstrap/bootstrap.js"></script>
     <!-- Load environmental scripts such as live reload -->
     {% for script in environmentalScripts %}
-    {{script}}
+    {% autoescape false %}{{script}}{% endautoescape %}
     {% endfor %}
     <script type="application/javascript">
     const areCookiesEnabled = () => {
@@ -176,4 +176,4 @@
 
 </body>
 
-</html>
\ No newline at end of file
+</html>
```

nbarrera committed:
En el archivo `app/views/login.html`

cambio para que el script ese no aparezca escapeado y se vea en el html en la pantalla de login

nbarrera committed:
arregla la pagina de "enviar memo"

para que funcione despues de habilitar csrf... el formulario no tenia el hidden input y entonces al submitear daba un error

nbarrera committed:
arregla loguearse con los usuarios que ya vienen

despues de haber hecho el cambio de hashear la password y comparar hashes

nbarrera committed:
add secure and maxAge for cookie

angelraver committed:
Merge pull request #2 from nbarrera/cambios-despues-de-test

Cambios despues de test
